source/_integrations/smappee.markdown

```diff
@@ -39,3 +46,9 @@ client_secret:
   required: true
   type: string
 {% endconfiguration %}
+
+<div class='note'>
+
+If Home Assistant is running in an non local environment, you must make sure that you have configured the `internal_url` in your [configuration](/docs/configuration/basic) in order to complete to OAuth2 configuration flow.
```

Comment:
I'm sorry, but this still is not correct. The internal URL doesn't need to be configured. Furthermore, the internal URL configuration option is hidden by default and only shown to users who have enabled advanced mode on their profile.

As I mentioned in my last comment. If no internal URL is set by the user, the IP address is used as the internal URL instead.

Reply:
I see, but currently our users cannot manage their app devices yet (i.e. setting or changing the redirect URL). Normally this should not be a problem (the detected IP address will work), except for a docker or other cloud environment.

Reply:
We require Home Assistant in containerized environments (e.g., Docker) to run on the host network. So that should not be a problem. As for the Home Assistant Cloud, that is considered an external URL and does not apply here.

Reply:
Running the Home Assistant containers on Windows is not a supported platform by the Home Assistant project.

Reply:
@frenck same unreachable callback URL on a Mac (docker)

Reply:
This is expected, since BSD-based systems (like macOS), isn't supported either. Our documentation does not take those into account.